### PR TITLE
[refactor] Remove import taichi in common_ops.py

### PR DESCRIPTION
**Pull request by lin-hitonami**
Related issue = #3782 

In `ops.py`, a snode-related function introduced the dependence of `Matrix` to the file, which caused circular import. So, I moved the snode-related functions to `snode.py`.

### CODE REVIEW

**Comment by taichi-gardener** [comment]
current commit hash: d993fd0e64a8a2f18e57736ebacf365cbfbb7369

master commit hash: 3818ba3142288752ebdd8c57bd47e92501aafcac

| Examples(CUDA) | ad_gravity | comet | cornell_box | fem128 | fem99 | fractal | game_of_life | inital_value_problem | mpm128 | mpm3d | mpm88 | mpm99 | mpm_lagrangian_forces | nbody | odop_solar | pbf2d | sdf_renderer | simple_uv | taichi_logo | vortex_rings |
|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|
| kernel_elapsed_time ( current/master ) | -0.0416% | -1.3471% | -0.3228% | -0.1157% | +0.8167% | -0.2655% | +2.9026% | -0.7224% | +0.7061% | +0.8582% | -0.6154% | +0.8595% | +0.4058% | -0.527% | +1.6826% | -0.0916% | +0.0939% | -0.0161% | -0.789% | -0.0258% |
| end2end_time ( current/master ) | -0.3676% | -0.7893% | -0.3862% | -0.1592% | +0.5727% | +0.4706% | -0.309% | -0.0571% | +4.0837% | -0.6098% | -0.2855% | +0.3724% | -1.6634% | -1.1613% | -1.9815% | -1.4026% | +0.327% | -0.0367% | -0.0154% | -0.6133% |


| benchmarks fluctuation threshold | 5.0% |
|:--:|:--:|
| kernel improvement | None |
| kernel regression | None |
| end2end improvement | None |
| end2end regression | None |

File: python/taichi/lang/common_ops.py [hunk 1]
```diff
@@ -1,135 +1,135 @@
-import taichi as ti
+from taichi.lang import ops as ti_ops
```
Comment (ailzhang) [inline python/taichi/lang/common_ops.py:1]: OOC: why not from `taichi.lang` import ops? ;)

File: python/taichi/lang/ops.py [hunk 2]
```diff
@@ -27,7 +37,8 @@ def stack_info():
 
 
 def is_taichi_expr(a):
-    return isinstance(a, Expr)
+    from taichi.lang.expr import Expr as ti_Expr  # pylint: disable=C0415
```
Comment (ailzhang) [inline python/taichi/lang/ops.py:40]: nit: no need to alias here?

File: python/taichi/lang/ops.py [hunk 3]
```diff
@@ -5,16 +5,26 @@
...
-from taichi.lang.expr import Expr, make_expr_group
 from taichi.lang.field import Field
 from taichi.lang.snode import SNode
 from taichi.lang.util import cook_dtype, is_taichi_class, taichi_scope
 
 unary_ops = []
 
 
+def Expr(*args, **kw):
+    from taichi.lang.expr import Expr as ti_Expr  # pylint: disable=C0415
```
Comment (ailzhang) [inline python/taichi/lang/ops.py:18]: ditto: is the the alias required?

File: python/taichi/lang/ops.py [hunk 4]
```diff
@@ -5,16 +5,26 @@
...
+    return ti_Expr(*args, **kw)
+
+
+def make_expr_group(*args):
+    from taichi.lang.expr import \
```
Comment (ailzhang) [inline python/taichi/lang/ops.py:23]: ditto

**Review by ailzhang — COMMENTED** [review]
Nice work!

**Review by ailzhang — APPROVED** [review]
LGTM!

**Review by strongoier — APPROVED** [review]
LGTM!

**Pull request merged**